### DISCUSSION TITLE
Does Blaze support FetchSize when returning results as a stream?

### DISCUSSION THREAD

**Discussion by isaac0977** [Blazebit/blaze-persistence#1527] [Q&A] [answered]
From my understanding, if I have this following code using Hibernate, I can stream 10 objects at a time. 
```
sessionFactory.openStatelessSession(DataSourceUtils.getConnection(myDataSource))
            .createNativeQuery("my_query", MyEntity.class)
            .setFetchSize(10)
            .stream();
```

I'm trying to achieve a similar result using Blaze, I have the following codeblock:

```
CriteriaBuilderFactory.create(entityManager,  MyEntity.class);
                .where("myEntityField").le(someValue)
                .getResultStream();
```

But I wasn't able to find any corresponding method to (setFetchSize), is this something that could be achieved using Blaze 1.6.6?

**Comment by jwgmeligmeyling** [comment, marked as answer]
I don't think we have an ability to set the fetch size, we just rely on the configured defeault for Hibernate.

Perhaps something like `criteriaBuilder.getQuery().unwrap(Query.class).setFetchSize(10).stream()` does the job for you?

**Reply by beikov** [reply]
Yeah, that's the best we can offer right now, but we can abstract this. I created https://github.com/Blazebit/blaze-persistence/issues/1528 for this.

**Reply by isaac0977** [reply]
Thank you, that will work for me for now.
